### PR TITLE
Safeguard logger against crashes in case caller doesn't send NSUUID type correlationId

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

**Comment by antrix1989** [comment]
Is it the only place where it can crash? Maybe we should put assert here, so it will raise all our internal issues with passing wrong class type. If the wrong class type was provided from our public API, we can verify it there and just ignore it.

File: IdentityCore/src/logger/MSIDLogger.m [hunk 1]
```diff
@@ -126,7 +126,7 @@ - (void)logWithLevel:(MSIDLogLevel)level
             
             NSString *correlationIdStr = @"";
             
-            if (correlationId)
```
Comment (jasoncoolmax) [inline IdentityCore/src/logger/MSIDLogger.m]: I think at least we should log a warning about wrong type being passed, if we don't want to assert here :)

**Comment by oldalton** [comment]
@antrix1989, @jasoncoolmax  I don't think assert would help in all cases. This is mainly for cases where we have dynamic data - e.g. last time we had a crash in ADAL for this issue was when broker was sending correlation ID to ADAL. In all our testing broker wasn't sending correlationId, so everything was working fine. We released a new version of broker and started sending correlationId that made ADAL to crash. 

So, to sum up, I think both assert and a warning is a good idea, but we still need to not crash when we see unexpected data, because the data can be coming from outside and our testing doesn't cover everything.

**Comment by antrix1989** [comment]
Agree, I didn't mean to remove this logic but add the assert on top of it so we can catch this at least for our internal code.
nit: show we verify class type when we get data outside of sdk? For example in PublcClientApplication and broker response handler? Currently, this fix doesn't prevent us from crashing if we use this method in other places.
